### PR TITLE
CT 1443 remove root path

### LINKED ISSUE #6171: [CT-1443] Remove 'root_path' from most node definitions
We are moving towards enabling manifests to be portable. Currently we include the 'root_path' of the project in every node, which is duplicative and non-portable. In addition, this field is hardly used.

Remove root_path from the node class definitions.

Note: because we don't load seed csv file contents, we do use the root_path to load the seed file when the seeds command is run. For now, we will include the root_path in seed nodes, but longer term we need to come up with a solution for this, whether making it possible to get the root_path for every project in the 'load_agate_table' context method or some other solution.

This is done as part of the structured logging effort, as cleanup of the node attributes that need to be included in the contracts for node information.

### PR DESCRIPTION
**Pull request by gshank**
resolves #6171

### Description

As part of cleanup for the structured logging effort, remove the 'root_path' field from most nodes.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: schemas/dbt/run-results/v4.json [hunk 1]
```diff
@@ -37,12 +37,12 @@
         },
         "dbt_version": {
           "type": "string",
-          "default": "1.2.0a1"
+          "default": "1.4.0a1"
```
Comment (emmyoop) [inline schemas/dbt/run-results/v4.json:40]: It looks like only the default and description fields are changing.  Do we really need to commit this?

File: schemas/dbt/manifest/v7.json [hunk 2]
```diff
@@ -366,9 +365,6 @@
         "package_name": {
           "type": "string"
         },
-        "root_path": {
-          "type": "string"
-        },
```
Comment (emmyoop) [inline schemas/dbt/manifest/v7.json:371]: Do we need to bump the schema to v8 instead of modifying?

**Review by emmyoop — CHANGES_REQUESTED** [review]
A few schema comments

**Comment by gshank** [comment]
Yes, we did need to bump the schema to v8. Have done that work.

**Review by emmyoop — CHANGES_REQUESTED** [review]
The run-results schema is just updating defaults and descriptions so `run-results/v4.json` should be removed from the PR.

**Comment by gshank** [comment]
Yeah, I've removed that run-results file a couple of times and then I run the script again... Gone now, I hope.

**Review by emmyoop — APPROVED** [review]

**Pull request merged**